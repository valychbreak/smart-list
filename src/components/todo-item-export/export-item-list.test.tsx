import { shallow } from "enzyme";
import ExportItem from "./export-item";
import ExportItemList from "./export-item-list";

type OverridingProps = {
    exportItems: ExportItem[];
};

function createExportItem(id: number, generalName: string) {
    return new ExportItem(
        id,
        generalName,
        1,
        true,
        null,
        null,
        null,
        null,
        null
    );
}

describe("ExportItemList", () => {
    const exportItemList = ({ ...overridingProps }: OverridingProps) => (
        <ExportItemList onEdit={() => {}} {...overridingProps} />
    );

    it("should display all todo items", () => {
        const exportItems = [
            createExportItem(1, "Milk"),
            createExportItem(2, "Bread"),
        ];

        const wrapper = shallow(exportItemList({ exportItems }));

        expect(wrapper).toMatchSnapshot();
    });
});
