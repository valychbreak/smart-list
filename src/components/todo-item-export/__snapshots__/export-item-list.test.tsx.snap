// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportItemList should display all todo items 1`] = `
<Styled(MuiBox)
  padding={1}
>
  <ExportItemView
    exportItem={
      ExportItem {
        "category": null,
        "generalName": "Milk",
        "id": 1,
        "isBought": true,
        "productPrice": null,
        "purchasedPrice": null,
        "purchasedStoreInternal": null,
        "quantity": 1,
        "targetProduct": null,
      }
    }
    key="1"
    onEdit={[Function]}
  />
  <ExportItemView
    exportItem={
      ExportItem {
        "category": null,
        "generalName": "Bread",
        "id": 2,
        "isBought": true,
        "productPrice": null,
        "purchasedPrice": null,
        "purchasedStoreInternal": null,
        "quantity": 1,
        "targetProduct": null,
      }
    }
    key="2"
    onEdit={[Function]}
  />
</Styled(MuiBox)>
`;
